repos/system_upgrade/common/actors/scanthirdpartytargetpythonmodules/libraries/scanthirdpartytargetpythonmodules.py

```diff
@@ -0,0 +1,189 @@
+    rpms_to_check = defaultdict(list)
+    third_party_unowned_files = []
+
+    for path in system_paths:
+        if not Path(path).is_dir():
+            continue
+        for file in find_python_related(path):
+            if _should_skip_file(file):
+                continue
+            owner = rpm_files.get(str(file))
+            if owner:
+                rpms_to_check[owner].append(file)
+            else:
+                third_party_unowned_files.append(str(file))
+
+    return rpms_to_check, third_party_unowned_files
+
+
+def identify_unsigned_rpms(rpms_to_check):
+    """
+    Identify which RPMs are third-party (not signed by the distribution).
+
+    :param rpms_to_check: Dictionary mapping RPM names to list of their files
+    :return: Tuple of (third_party_rpms, third_party_files) where:
+             - third_party_rpms is a list of third-party RPM package names
+             - third_party_files is a list of files from third-party RPMs
+    """
+    third_party_rpms = []
+    third_party_files = []
+
+    for rpm_name, files in rpms_to_check.items():
+        if not has_package(DistributionSignedRPM, rpm_name):
+            third_party_rpms.append(rpm_name)
+            api.current_logger().warning(
+                'Found Python files from non-distribution RPM package: {}'.format(rpm_name)
+            )
+            third_party_files.extend([str(f) for f in files])
+
+    return third_party_rpms, third_party_files
+
+
+def process():
+    """
+    Main function to scan for third-party Python modules/RPMs on the target system.
+
+    This function:
+    1. Validates the target RHEL version and Python interpreter
+    2. Scans system paths for Python files
+    3. Identifies third-party RPMs and modules
+    4. Produces a message if any third-party modules/RPMs are detected
+    """
+    target_version = get_target_major_version()
+    target_python = get_python_binary_for_rhel(target_version)
+
+    if not target_python:
+        api.current_logger().info(
+            "RHEL version {} is not supported for third-party Python modules scanning, "
+            "skipping check.".format(target_version)
+        )
+        return
+
+    if not is_target_python_present(target_python):
+        api.current_logger().info(
+            "Target Python interpreter {} is not installed on the source system, "
+            "skipping check of 3rd party python modules.".format(target_python)
+        )
+        return
+    system_paths = get_python_sys_paths(target_python)
+    rpm_files = identify_files_of_pypackages(system_paths[1:])
+
+    rpms_to_check, third_party_unowned_files = scan_python_files(system_paths[1:], rpm_files)
+
+    third_party_rpms, rpm_owned_files = identify_unsigned_rpms(rpms_to_check)
+
+    # Combine all third-party files (unowned + from third-party RPMs)
+    all_third_party_files = third_party_unowned_files + rpm_owned_files
+
+    if third_party_rpms or all_third_party_files:
+        api.current_logger().warning(
```

Comment:
Again, this is a situation we are prepared for, use `debug` or `info`
```suggestion
        api.current_logger().debug(
```